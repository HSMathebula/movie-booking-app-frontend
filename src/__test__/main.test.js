import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Main from '../components/movies/main';

const movies = [
  {
    description: 'this is a test movie',
    id: 1,
    title: 'Star wars',
    picture:
      'https://image.shutterstock.com/image-vector/hawaii-beach-bar-retro-sign-600w-2156338567.jpg',
  },
];

describe('testing the rendering of the main page', () => {
  it('', () => {
    const { container } = render(
      <Provider store={store}>
        <Main movies={movies} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
