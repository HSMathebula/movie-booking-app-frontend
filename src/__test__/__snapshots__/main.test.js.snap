// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the rendering of the main page  1`] = `
<div>
  <div
    class="container pt-5"
  >
    <div
      class="text-center mb-5"
    >
      <h1
        class="fw-bold"
      >
        Latest Movies
      </h1>
      <p>
        Most watched latest movies on Movie Bisko.
      </p>
    </div>
    <div
      class="row"
    >
      <div
        class="col-1 d-flex align-items-center d-none d-md-flex"
      >
        <div
          class="arrow-container "
        >
          <svg
            class="prev-button"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M247 65.16L32.34 440.8l61.79-35.7L247 137.6zm18 .38V137l158.3 271.3 62.7 36.1C412.2 318.2 338.6 191.8 265 65.54zM415.4 424.5l-321.3 1.4-62.72 36.2 445.82-1.9z"
            />
          </svg>
        </div>
      </div>
      <div
        class="col col-md-10"
      >
        <div
          class="row"
        />
      </div>
      <div
        class="col-1 d-flex align-items-center d-none d-md-flex"
      >
        <div
          class="arrow-container "
        >
          <svg
            class="next-button"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M247 65.16L32.34 440.8l61.79-35.7L247 137.6zm18 .38V137l158.3 271.3 62.7 36.1C412.2 318.2 338.6 191.8 265 65.54zM415.4 424.5l-321.3 1.4-62.72 36.2 445.82-1.9z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
